### PR TITLE
statistics: use PrevLastVersion to avoid error.

### PR DESCRIPTION
**Pull request by hanfei1991**
Currently, we use startTS of a txn as the version of a stats table. There may be an error like this:
txn1 with version 2 write firstly.
TiDB update it, and the latest udpate version is 2.
then txn2 with version 1 write.
TiDB will never get it with the version 2.

So we choose to get the table stats which version is greator than the version before **last two lease**.

If the commitTime - startTS is less than a lease time(always a few seconds), we can always read it.

### CODE REVIEW

File: statistics/statscache_test.go [hunk 1]
```diff
@@ -242,10 +242,65 @@ func (s *testStatsCacheSuite) TestDDL(c *C) {
...
+	// A smaller version write, and we can still read it.
+	testKit.MustExec("update mysql.stats_meta set version = 1 where table_id = ?", tableInfo2.ID)
+	h.Update(is)
+	statsTbl2 := h.GetTableStats(tableInfo2)
+	c.Assert(statsTbl2.Pseudo, IsFalse)
+
+	testKit.MustExec("insert t1 values(1,2)")
+	testKit.MustExec("analyze table t1")
+	testKit.MustExec("update mysql.stats_meta set version = 4 where table_id = ?", tableInfo1.ID)
+	h.Update(is)
```
Comment (coocood) [inline statistics/statscache_test.go]: Asset last version and last two version makes the test more readable.

File: statistics/statscache_test.go [hunk 2]
```diff
@@ -242,10 +242,65 @@ func (s *testStatsCacheSuite) TestDDL(c *C) {
...
+	testKit.MustExec("update mysql.stats_meta set version = 3 where table_id = ?", tableInfo2.ID)
+	h.Update(is)
+	statsTbl2 = h.GetTableStats(tableInfo2)
+	c.Assert(statsTbl2.Count, Equals, int64(1))
+
+	testKit.MustExec("insert t2 values(1,2)")
+	testKit.MustExec("analyze table t2")
+	// A smaller version write, and we cannot read it. Because at this time, lastTwo Version is 4.
+	testKit.MustExec("update mysql.stats_meta set version = 3 where table_id = ?", tableInfo2.ID)
+	h.Update(is)
```
Comment (coocood) [inline statistics/statscache_test.go]: Ditto

File: statistics/statscache_test.go [hunk 3]
```diff
@@ -242,10 +242,65 @@ func (s *testStatsCacheSuite) TestDDL(c *C) {
...
+	testKit.MustExec("update mysql.stats_meta set version = 4 where table_id = ?", tableInfo1.ID)
+	h.Update(is)
+	statsTbl1 = h.GetTableStats(tableInfo1)
+	c.Assert(statsTbl1.Count, Equals, int64(1))
+
+	testKit.MustExec("insert t2 values(1,2)")
+	testKit.MustExec("analyze table t2")
+	// A smaller version write, and we can still read it.
+	testKit.MustExec("update mysql.stats_meta set version = 3 where table_id = ?", tableInfo2.ID)
+	h.Update(is)
```
Comment (coocood) [inline statistics/statscache_test.go]: Ditto

File: statistics/statscache_test.go [hunk 4]
```diff
@@ -242,10 +242,65 @@ func (s *testStatsCacheSuite) TestDDL(c *C) {
...
+
+	testKit.MustExec("create table t2 (c1 int, c2 int)")
+	testKit.MustExec("analyze table t2")
+	is = do.InfoSchema()
+	tbl2, err := is.TableByName(model.NewCIStr("test"), model.NewCIStr("t2"))
+	c.Assert(err, IsNil)
+	tableInfo2 := tbl2.Meta()
+	// A smaller version write, and we can still read it.
+	testKit.MustExec("update mysql.stats_meta set version = 1 where table_id = ?", tableInfo2.ID)
+	h.Update(is)
```
Comment (coocood) [inline statistics/statscache_test.go]: So if we load the stats for the second time, every stats will be reloaded?

Comment (hanfei1991) [inline statistics/statscache_test.go]: Yeah, the stats will be loaded 2 times even if it was not updated, but after #3059 , The histogram will not be loaded multiple times.

**Comment by zimulala** [comment]
LGTM

**Comment by zimulala** [comment]
LastTwoVersion->PrevLastVersion ?

**Comment by hanfei1991** [comment]
@zimulala I will fix it.

**Pull request merged**